### ISSUE TITLE
Crash on 0.20.1 with Ampere

### ISSUE THREAD

**Issue by jorge-barroso** [trexminer/T-Rex#429]
The miner will crash randomly at some point while mining, not lasting more than 1-2 minutes active with version 0.20.1

System:
CPU: Ryzen 5800X
MEM: 32GB 3600Mhz DDR4
Mobo: X570 (GPU using PCIe 4.0 16x)
OS: Manjaro w/ Linux 5.11.14
GPU: Zotac RTX 3070 Twin Edge w/ driver 460.67
Command line (the part in square brackets is something I have tried putting on and off in different combinations): t-rex -a ethash -o stratum+tcp://eu-eth.hiveon.net:4444 -u 0x... -p x -w RTX_3070 --kernel 5 [--api-bind-telnet 0 --api-bind-http 0 -i 12.75]

Error log example:
20210417 12:08:32 [ OK ] 6/6 - 59.78 MH/s, 18ms ... GPU #0
20210417 12:08:40 TREX: eyJhIjoiOWlTTFVBdjhzaTFydXR3c0pPKzVGdmFTVUFFYktjbGZOaGlKVnQreFB0ZUhwYkNGUnEwNmJmR3BESkJnOFVqcWRGTGlIaE5VSWRWMDdRQi9qazI3VDlSMnlkQTJoU1UzT1FiRzMzbFUvRWs9IiwiYiI6Im9HN2FseTFJSUdMM2xBaVRBRXh4Vnc9PSJ9Cg==
20210417 12:08:40 WARN: Miner is going to shutdown...
20210417 12:08:40 Main loop finished. Cleaning up resources...
20210417 12:08:46 T-Rex finished.
Killed

I have downloaded a local copy of 0.20.0 and it is stable with no crashes so far in over a day

**Comment by trexminer** [comment]
Thanks for reporting the issue. We'll fix it in the next version.

**Comment by sniperlucian** [comment]
same here, was working fine and suddenly start to crash 

RTX2080TI 

./t-rex -a mtp  -o stratum+tcp://firo-eu.mintpond.com:3000 -u  xxxxxxxxxxxxxxxxxxxx   -p 0

**Comment by trexminer** [comment]
@jorge-barroso When you have a moment, could you please give this new test build a try and let me know if it fixes the issue?
https://www.dropbox.com/s/j4c02wtv9aacoid/t-rex-0.20.1-linux.tar.gz?dl=1

**Comment by jorge-barroso** [comment]
> @jorge-barroso When you have a moment, could you please give this new test build a try and let me know if it fixes the issue?
> https://www.dropbox.com/s/j4c02wtv9aacoid/t-rex-0.20.1-linux.tar.gz?dl=1

@trexminer I'm testing it now and it has been running for an hour with no issues so I'd say it's fixed for me

**Comment by renaatski** [comment]
Can there be shared more about this exact problem?   Ever since I'm running 0.20.1 on hiveOS I'm getting strange kernel lockups... requiring a hard reboot. This usually happens within a few minutes, sometimes longer (like withing the hour)

Reverted to 0.19.14-cuda111 , and have been running great for 3h30m  , no rejects

Is this related/similar to the OP?

**Comment by ireun** [comment]
Same here on Windows, please fix :(

**Comment by trexminer** [comment]
> Same here on Windows, please fix :(

This error is Linux specific. Please open a new issue for Windows.

**Comment by trexminer** [comment]
> Can there be shared more about this exact problem? Ever since I'm running 0.20.1 on hiveOS I'm getting strange kernel lockups... requiring a hard reboot. This usually happens within a few minutes, sometimes longer (like within the hour)
> 
> Reverted to 0.19.14-cuda111 , and have been running great for 3h30m , no rejects
> 
> Is this related/similar to the OP? I don't have a log entry or anything, just a hard freeze.

No, what you describe sounds unrelated to this issue which is just an OS trying to inject libraries into the miner
